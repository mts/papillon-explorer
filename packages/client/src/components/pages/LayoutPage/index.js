import React from 'react';
import { Link } from 'react-router-dom';
import { startPageRoute } from '../../../routes'

const LayoutPage = () => {
  return (
    <div>
      <h2>Layout Page</h2>
      <div>
        Under construction ðŸ™‚
        <Link to={startPageRoute}> Back to Home Page</Link>
      </div>
    </div>
  )
}

export default LayoutPage;
