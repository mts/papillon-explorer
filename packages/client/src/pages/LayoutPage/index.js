import React from 'react'
import { Query } from 'react-apollo'
import { Link } from 'react-router-dom'
import { startPageRoute } from '../../routes'
import { PAGES } from '../../graphql/query'

const LayoutPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? (
          <div>
            <h2>{data.pages.pages.find(page => page.id === 'layout').name}</h2>
            <div>
              Under construction ğŸ™‚
              <Link to={startPageRoute}> Back to Home Page</Link>
            </div>
          </div>
        ) : null
      }
    </Query>
  )
}

export default LayoutPage
