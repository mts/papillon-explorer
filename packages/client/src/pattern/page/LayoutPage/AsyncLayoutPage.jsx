import React from 'react'

const LazyLayoutPage = React.lazy(() => import('./LayoutPage'))

export const AsyncLayoutPage = () => (
  <React.Suspense fallback={null}>
    <LazyLayoutPage />
  </React.Suspense>
)
