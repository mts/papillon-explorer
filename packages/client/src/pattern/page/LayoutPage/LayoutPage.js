import React from 'react'
import { Query } from 'react-apollo'
import { startPageRoute } from '../../../routes'
import { PAGES } from '../../../graphql/query'
import LayoutTemplate from '../../template/LayoutTemplate'

const LayoutPage = () => {
  return <Query query={PAGES}>{({ data }) => <LayoutTemplate data={data} startPageRoute={startPageRoute} />}</Query>
}

export default LayoutPage
