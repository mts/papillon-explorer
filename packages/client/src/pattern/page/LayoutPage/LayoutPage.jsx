import React from 'react'
import { Query } from 'react-apollo'
import { PAGES } from '../../../graphql/query'
import { LayoutTemplate } from '../../template/LayoutTemplate'

export const LayoutPage = () => {
  return (
    <Query query={PAGES}>
      {({ data }) =>
        data.pages && data.pages.pages ? <LayoutTemplate page={data.pages.pages.find(page => page.id === 'layout')} /> : null
      }
    </Query>
  )
}
