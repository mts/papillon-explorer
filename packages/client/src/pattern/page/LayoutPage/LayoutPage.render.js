import React from 'react'
import { AsyncLayoutPage } from './AsyncLayoutPage'
import { LayoutPage } from './LayoutPage'

export const layoutPageAsync = <AsyncLayoutPage />
export const layoutPageDefault = <LayoutPage />
