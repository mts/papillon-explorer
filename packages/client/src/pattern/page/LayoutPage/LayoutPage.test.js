import React from 'react'
import { layoutPageDefault } from './LayoutPage.render'

jest.mock('@apollo/react-hooks', () => {
  return {
    useQuery: () => {
      return {
        data: {
          pages: {
            pages: [{ id: 'layout', name: 'some-name' }],
          },
        },
      }
    },
  }
})

jest.mock('react-router-dom', () => ({ Link: () => <div>some link</div> }))

describe('<LayoutPage />', () => {
  describe('Snaphot', () => {
    test('must match layoutPageDefault', () => {
      expect(global.renderToJSON(layoutPageDefault)).toMatchSnapshot()
    })
  })
})
