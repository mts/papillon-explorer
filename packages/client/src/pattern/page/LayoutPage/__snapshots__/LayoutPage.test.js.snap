// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutPage /> Snaphot must match layoutPageDefault 1`] = `
<div>
  <h2>
    some-name
  </h2>
  <div>
    Under construction
     
    <span
      aria-label="emoji"
      role="img"
    >
      ðŸ™‚
    </span>
    <div>
      some link
    </div>
  </div>
</div>
`;
